packages/react-experiments/src/components/UnifiedPicker/UnifiedPicker.tsx

```diff
@@ -296,11 +296,20 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
       // Allow the caller to handle the key down
       onKeyDown?.(ev);
 
-      // Handle copy if focus is in the selected items list
       // This is a temporary work around, it has localization issues
       // we plan on rewriting how this works in the future
-      // eslint-disable-next-line deprecation/deprecation
-      if (ev.ctrlKey && ev.which === KeyCodes.c) {
+      /* eslint-disable deprecation/deprecation */
+      const isDel = ev.which === KeyCodes.del;
+      const isCut = (ev.shiftKey && isDel) || (ev.ctrlKey && ev.which === KeyCodes.x);
+      const isBackspace = ev.which === KeyCodes.backspace;
+      const isCopy = ev.ctrlKey && ev.which === KeyCodes.c;
+      /* eslint-enable deprecation/deprecation */
+      const needToCopy = isCut || isCopy;
+      const needToDelete =
+        (isBackspace && selectedItems.length > 0) || ((isCut || isDel) && focusedItemIndices.length > 0);
```

Comment:
I think the focusedItemIndices check needs to be removed - backspace should remove the last one if nothing is selected

Reply:
Ah, this is related to the confusing thing I investigated yesterday. The variable names seem to be incorrect, but I didn't change them (yet):

`selectedItems` always contains **all** the items in the well. So this check is necessary for the backspace case to ensure we don't try to delete when nothing exists to delete. (Or that seems to be the reasoning... I haven't yet validated whether we _actually_ need the check. I just preserved the pre-existing check when refactoring)

`focusedItemIndicies` contains the **selected** item indicies. So I added this check to make sure we only Cut/Delete when something is _selected_. Should change the name so this is clear.

`selection` contains the selected items.

While Backspace should delete the last item if nothing is selected, that is not true for Cut/Del scenarios. If nothing is selected, then Cut/Del should do nothing. Currently it's not possible to put focus between two recipients _(so something is always selected when focus is in the middle)_, but in the future Delete should remove the recipient **after** the insertion point _if nothing is selected_.

Reply:
Yep. Ah yes I read this wrong with the parenthesis

Reply:
I'm looking at the `selectedItems` thing more, and it seems some of the code really intends for that to mean "selected" items _(and not necessarily always mean "all" items)_, but whatever concept of selection is intended doesn't seem to match the user's concept of selection... so it's very confusing to me.

Reply:
Yeah selectedItems should really be called list items or at least selectableItems